Updating the backup-restore docs. (#1032)
gardener/etcd-backup-restore: README.md

# Etcd-Backup-Restore
[![REUSE status](https://api.reuse.software/badge/github.com/gardener/etcd-backup-restore)](https://api.reuse.software/info/github.com/gardener/etcd-backup-restore)
[![Go Report Card](https://goreportcard.com/badge/github.com/gardener/etcd-backup-restore)](https://goreportcard.com/report/github.com/gardener/etcd-backup-restore)
[![GoDoc](https://godoc.org/github.com/gardener/etcd-backup-restore?status.svg)](https://godoc.org/github.com/gardener/etcd-backup-restore)

> Etcd-backup-restore is collection of components for backuping and restoring the [etcd](https://github.com/etcd-io/etcd/). It also validates the etcd's data directory and auto-triggers restoration when needed.

Reviewer: ```suggestion
Etcd-backup-restore is collection of components for backing up and restoring an [etcd](https://github.com/etcd-io/etcd/) cluster. It also validates the etcd's data directory and auto-triggers restoration when needed.
```

> Etcd-backup-restore is collection of components for backing up and restoring an [etcd](https://github.com/etcd-io/etcd/) cluster. It also validates the etcd's data directory and auto-triggers restoration when needed.

Reviewer: ```suggestion
Etcd-backup-restore is a collection of components for backing up and restoring an [etcd](https://github.com/etcd-io/etcd/) cluster. It also validates the etcd's data directory and auto-triggers restoration when needed.
```

etcd-backup-restore currently provides the following capabilities (the list is not comprehensive):

>   - **Backups**: etcd database snapshots(a.k.a full snapshots) and incremental snapshots are taken regularly, compressed and stored in the configured object storage provider.

Reviewer: ```suggestion
  - **Backups**: etcd database snapshots (a.k.a full snapshots) and incremental snapshots are taken regularly, compressed and stored in the configured object storage provider.
```

>   - **Backups**: [etcd](https://github.com/etcd-io/etcd) database snapshots(a.k.a full snapshots) and incremental snapshots are taken regularly, compressed and stored in the configured object storage provider.

Reviewer: ```suggestion
  - **Backups**: etcd database snapshots(a.k.a full snapshots) and incremental snapshots are taken regularly, compressed and stored in the configured object storage provider.
```

We don't need to link to etcd-io github repo everytime we mention `etcd`

>   - **Data directory validation**: It does [etcd](https://github.com/etcd-io/etcd) data-directory validation before starting any etcd member.

Reviewer: ```suggestion
  - **Data directory validation**: It validates the etcd data directory before starting any etcd member.
```

Same reason as above

  - **Restoration**: In case of a database corruption for a single-member cluster it restores from latest set of snapshots (full & delta).
  - **Zero-downtime member recovery**: Replaces a failed member in a multi-member etcd cluster without any downtime.

>   - **Automated maintenance**: Act as watchDog to make etcd cluster lean and clean by running [maintenance](https://etcd.io/docs/v3.5/op-guide/maintenance/) operations such as defragmentation on a configured schedule.

Reviewer: ```suggestion
  - **Automated maintenance**: Runs database [maintenance](https://etcd.io/docs/v3.5/op-guide/maintenance/) operations such as defragmentation on a configured schedule.
```

Author: I would like to avoid LLM generated sentences sometimes, that's why I choose above wording.

Reviewer: But it's grammatically incorrect and also doesn't add value to the reader with phrases like `lean and clean`, so it makes sense to remove it and talk about precisely what etcd-backup-restore offers.

>   - **Secondary backups**: Uploading the backups to secondary object store (if configured).

Reviewer: ```suggestion
  - **Secondary backups**: In addition to primary backup storage, it uploads backups to a secondary object store, if configured, for higher backup resilience.
```

  - **Metrics**: Exposes metrics to monitor the health and status of etcd-backup-restore operations.

> For detailed documentation, see our `/docs` folder. Please find the [index](docs/README.md) here.

Reviewer: ```suggestion
For detailed documentation, head over to [`/docs`](docs/README.md).
```

> Please check this for [Getting started](deployment/getting_started.md) and for detailed documentation please find the [index](docs/README.md) here.

Reviewer: Add a separate section for this, like 
```suggestion
## Getting Started

Please check the [Getting started](docs/deployment/getting_started.md) guide for instructions on installation and basic usage.

For detailed documentation, head over to the [docs](docs/README.md).
```

(includes link fix as well)

## Contributions

> If you wish to contribute then please see our [contributor guidelines](development/contribution.md).

Reviewer: ```suggestion
If you wish to contribute then please see our [contributor guide](docs/development/contribution.md).
```

## Feedback and Support

We always look forward to active community engagement. Please report bugs or suggestions on how we can enhance `etcd-backup-restore` on [GitHub Issues](https://github.com/gardener/etcd-backup-restore/issues).

## License

> Release under [Apache-2.0](https://github.com/gardener/etcd-backup-restore/blob/master/LICENSE) license.

Reviewer: ```suggestion
Released under [Apache-2.0](https://github.com/gardener/etcd-backup-restore/blob/master/LICENSE) license.
```

Author: I picked from druid: https://github.com/gardener/etcd-druid#license , release is written

Author: It should be `Release` imo.

Reviewer: As a sentence, you would say `This was released under this license` or `this is maintained under this license`. So past tense makes sense here

Author: present tense not past 
> this is maintained under this license

Author: anyway, I rephrase to `This project is maintained under [Apache-2.0](...) license.`
